src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEUtilities.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection.Metadata;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE;
+
+internal static class PEUtilities
+{
+    internal static void DeriveUseSiteInfoFromCompilerFeatureRequiredAttributes(ref UseSiteInfo<AssemblySymbol> result, Symbol symbol, PEModuleSymbol module, EntityHandle handle, CompilerFeatureRequiredFeatures allowedFeatures, MetadataDecoder decoder)
```

Comment:
>UseSiteInfo<AssemblySymbol> [](http://example.com/codeflow?start=84&length=27)

I think usage of this type adds an unnecessary complexity, consumers of this API are getting more complicated because of this too. We simply could return a DiagnosticInfo object.